deploy/backup/copy-backup.sh

```diff
@@ -68,58 +76,84 @@ enable_logging() {
     fi
 }
 
-check_input() {
-    local backup_dest=$1
-    local dest_dir=$2
-
-    echo
-    if [ -z "$backup_dest" ] || [ -z "$dest_dir" ]; then
-        usage
-    fi
+check_input_namespace() {
+    local namespace=${1}
 
-    if [ ! -e "$dest_dir" ]; then
-        mkdir -p "$dest_dir"
+    if [ -n "$namespace" ]; then
+        ctrl="$ctrl -n $namespace"
     fi
+}
 
-    if [ "${backup_dest:0:4}" = "pvc/" ]; then
-        if ! $ctrl get "$backup_dest" 1>/dev/null; then
-            printf "[ERROR] '%s' PVC doesn't exists.\n\n" "$backup_dest"
-            usage
-        fi
-    elif [ "${backup_dest:0:5}" = "s3://" ]; then
-        env -i $CREDENTIALS $xbcloud get ${backup_dest} xtrabackup_info 1>/dev/null
-    else
-        usage
-    fi
+check_input_destination() {
+  local backup_dest=$1
+  local dest_dir=$2
+
+  if [ -z "$backup_dest" ] || [ -z "$dest_dir" ]; then
+      usage
+  fi
+
+  if [ ! -e "$dest_dir" ]; then
+      mkdir -p "$dest_dir"
+  fi
+
+  if [ "${backup_dest:0:4}" = "pvc/" ]; then
+      if ! $ctrl get "$backup_dest" 1>/dev/null; then
+          printf "[ERROR] '%s' PVC doesn't exists.\n\n" "$backup_dest"
+          usage
+      fi
+  elif [ "${backup_dest:0:5}" = "s3://" ]; then
+      env -i $CREDENTIALS $xbcloud get ${backup_dest} xtrabackup_info 1>/dev/null
+  else
+      echo "Can't find $backup_dest backup"
+      usage
+  fi
+
+  if [ ! -d "$dest_dir" ]; then
+      printf "[ERROR] '%s' is not local directory.\n\n" "$dest_dir"
+      usage
+  fi
 
-    if [ ! -d "$dest_dir" ]; then
-        printf "[ERROR] '%s' is not local directory.\n\n" "$dest_dir"
-        usage
-    fi
 }
 
 start_tmp_pod() {
     local backup_pvc=$1
 
-    $ctrl delete pod/backup-access 2>/dev/null || :
-    cat - <<-EOF | $ctrl apply -f -
-		apiVersion: v1
-		kind: Pod
-		metadata:
-		  name: backup-access
-		spec:
-		      containers:
-		      - name: xtrabackup
-		        image: percona/percona-xtradb-cluster-operator:0.3.0-backup
-		        volumeMounts:
-		        - name: backup
-		          mountPath: /backup
-		      restartPolicy: Never
-		      volumes:
-		      - name: backup
-		        persistentVolumeClaim:
-		          claimName: ${backup_pvc#pvc/}
-	EOF
+    if [ -n "$namespace" ]; then
+      $ctrl delete pod/backup-access 2>/dev/null || echo "apiVersion: v1
```

Comment:
it is unclear why we should delete OR create backup-access pod.
The initial logic was - delete obsolete backup-access pod before the creation of backup-access pod.

Reply:
thanks, you have catched an issue here. So I've restored this `$ctrl delete pod/backup-access 2>/dev/null || :` command exactly as it was. Also put before `if` statement to avoid unnecessary duplicate. I believe this command is okay now. It does clean up before spinning a fresh pod for a backup. After changes, I've tested this and it worked well: `./copy-backup.sh cron-mattermost-clust-fs-pvc-20210320000009-372f8 /tmp/`date +%Y-%m-%d_%H-%M-%S` mm-p` on Mac (same will be on Linux).